lib/auth.js

```diff
@@ -99,12 +102,19 @@ export default class Auth {
      * @param {string} authObject.client_secret Client Secret from SFMC config
      * @param {number} authObject.account_id MID of Business Unit used for API Calls
      * @param {string} authObject.auth_url Auth URL from SFMC config
+     * @param {string} authObject.grant_type Grant Type. If present then access_token should be passed to the Auth object.
      * @param {string[]} [authObject.scope] Array of scopes used for requests
      * @param {object} options options for the SDK as a whole, for example collection of handler functions, or retry settings
      */
-    constructor(authObject, options) {
+    constructor(authObject, options, grant_type = GRANT_TYPE_CLIENT_CREDENTIALS) {
         if (!authObject) {
             throw new Error('authObject are required. see readme.');
+        } else if (grant_type && grant_type == GRANT_TYPE_AUTHORIZATION_CODE) {
```

Comment:
one thought here is that, since you are providing grant_type as a default, the additional check here is not needed since it will always have one. I would also prefer that since you only check for the error here, that this be included in the else if clause, rather than opening a new block to avoid nested if/else

Reply:
Thanks for the comment.
If I change the condition without nested if like the following
else if (grant_type == GRANT_TYPE_AUTHORIZATION_CODE && !authObject.access_token)

then the validation would fail asking for the client_id to be present if grant_type=authorization_code and access_token is present. 

Otherwise I have to add an exclusion of the grant_type!=authorization_code  in the other conditions.